2019/test/SuttersTest.java

```diff
@@ -0,0 +1,203 @@
+                            goldMineralPosition.setValue("LEFT");
+                            knockPath = knockLeftPath;
+                        } else if (goldMineralX > silverMineral1X && goldMineralX > silverMineral2X) {
+                            goldMineralPosition.setValue("RIGHT");
+                            knockPath = knockRightPath;
+                        } else {
+                            goldMineralPosition.setValue("CENTER");
+                            knockPath = knockCenterPath;
+                        }
+                    }
+                }
+            }
+        }
+        if (tfod != null) {
+            tfod.shutdown();
+        }
+
+    }
+
+    /*
+    protected void initializeMineralDetection() {
+        MineralDetectionTask mdTask = new MineralDetectionTask(this) {
+            public void handleEvent(RobotEvent e) {
+                MineralDetectionEvent event = (MineralDetectionEvent)e;
+                List<Recognition> updatedMinerals = event.minerals;
+                if (updatedMinerals != null) {
+                    numberOfMinerals.setValue(updatedMinerals.size());
+                    if (updatedMinerals.size() == 3) {
+                        int goldMineralX = -1;
+                        int silverMineral1X = -1;
+                        int silverMineral2X = -1;
+                        for (Recognition recognition : updatedMinerals) {
+                            if (recognition.getLabel().equals(LABEL_GOLD_MINERAL)) {
+                                goldMineralX = (int) recognition.getLeft();
+                            } else if (silverMineral1X == -1) {
+                                silverMineral1X = (int) recognition.getLeft();
+                            } else {
+                                silverMineral2X = (int) recognition.getLeft();
+                            }
+                        }
+                        if (goldMineralX != -1 && silverMineral1X != -1 && silverMineral2X != -1) {
+                            if (goldMineralX < silverMineral1X && goldMineralX < silverMineral2X) {
+                                goldMineralPosition.setValue("LEFT");
+                                knockPath = knockLeftPath;
+                            } else if (goldMineralX > silverMineral1X && goldMineralX > silverMineral2X) {
+                                goldMineralPosition.setValue("RIGHT");
+                                knockPath = knockRightPath;
+                            } else {
+                                goldMineralPosition.setValue("CENTER");
+                                knockPath = knockCenterPath;
+                            }
+                        }
+                    }
+                }
+
+            }
+        };
+        mdTask.init(telemetry, hardwareMap);
+        mdTask.setDetectionKind(MineralDetectionTask.DetectionKind.EVERYTHING);
+        this.addTask(mdTask);
```

Comment:
Take out the complete mess above and below, put this back in, and then add a method to the MineralDetectionTask called useExternalWebcam(String webcamName).  Call this before you call mdTask.init(....).

Then in initializeVurforia() if webcameName is not null, or the empty string then set parameters.cameraName as below using that name.